### DISCUSSION TITLE
groupBy duration parameter

### DISCUSSION THREAD

**Discussion by vdawg-git** [ReactiveX/rxjs#7402] [Q&A] [answered]
I want to limit the lifespan of streams created by [`groupBy`](https://rxjs.dev/api/index/function/groupBy) to prevent a memory leak.
For that the `duration` property seems to be the solution.
However, I cannot figure out how to use it. It is in the documentation, but not mentioned.

Can someone post a short pseudo-code example on how it is intended to be used? :)

Edit: I found a blog post which explains it quite well: I also found a blog-post where the author explained it quite well: 
https://eugene.coding.blog/rxjs-group-by

**Comment by voliva** [comment, marked as answer]
I've never used it, but you can see on the tests for an example: https://github.com/ReactiveX/rxjs/blob/master/packages/rxjs/spec/operators/groupBy-spec.ts#L91

It looks like you can return an observable that says when the group should be closed. On this example, it's using the same stream to say that it should get killed after the second emission.

If a new value comes through the stream for that group, a new group will be created.

**Reply by vdawg-git** [reply]
Thank you :)
